in-progress/7588-spartan-clusters.md

```diff
@@ -0,0 +1,301 @@
+# Configurable k8s clusters for testing and deploying networks
+
+|                      |                                                             |
+| -------------------- | ----------------------------------------------------------- |
+| Issue                | https://github.com/AztecProtocol/aztec-packages/issues/7588 |
+| Owners               | @just-mitch                                                 |
+| Approvers            | @PhilWindle @alexghr @charlielye                            |
+| Target Approval Date | 2024-08-16                                                  |
+
+
+## Executive Summary
+
+We will:
+- add a helm chart for deploying a configurable network.
+- add support for ad-hoc k8s clusters in CI and running e2e tests on them
+- add support for deploying a public network
+
+
+## Introduction
+
+To properly test a decentralized system, we need the ability to spin up networks with different topologies such as the number of nodes, validators, provers, pxes.
+
+Additionally, we need to test under simulated stress/attack conditions.
+
+Further, we need to be able to deploy these networks in a repeatable and automated way.
+
+> Kubernetes, also known as K8s, is an open source system for automating deployment, scaling, and management of containerized applications.
+
+And [Helm](https://helm.sh/) is "the package manager for Kubernetes".
+
+This allows us to define a network configuration in a helm chart and deploy it to a k8s cluster, setting variables on the fly such as the number of nodes, validators, machine resources, etc.
+
+K8s is also easy to use in CI via [kind](https://kind.sigs.k8s.io/).
+
+> kind is a tool for running local Kubernetes clusters using Docker container “nodes”.
+> kind was primarily designed for testing Kubernetes itself, but may be used for local development or CI.
+
+Further, we can use [chaos mesh](https://chaos-mesh.org/) to simulate network conditions such as node failures, latency, etc.
+
+Last, we can likely use the exact same chart to deploy a public network.
+
+### Why not docker compose?
+
+We use docker-compose already in some tests. The problem is that it is very difficult to test a network with more than a few nodes. It is also difficult to simulate network conditions. Last, we wouldn't be able to use the same tooling to deploy a public network.
+
+The thinking here is to use the same tooling that we use for production deployments to test our networks. This should result in less code and more confidence.
```

Comment:
To confirm: this would replace much of our tf templates then?

Reply:
Yes. TF can/should be used to set up the k8s cluster itself, and public networking infrastructure, but our software deployment configuration can live in helm.